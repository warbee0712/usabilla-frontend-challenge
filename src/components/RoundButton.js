import styled, { css } from 'styled-components'
import { size, fontSize } from 'styled-system'
import { focus } from '../sharedStyles'

const activeStyle = css`
  background-color: ${props => props.theme.colors.blue2};
  border-color: ${props => props.theme.colors.blue2};
  color: ${props => props.theme.colors.white};
  &:hover {
    background-color: ${props => props.theme.colors.blue1};
    border-color: ${props => props.theme.colors.blue1};
    box-shadow: ${props => props.theme.shadows.default};
    color: ${props => props.theme.colors.white};
  }
`

const RoundButton = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  color: ${props => props.theme.colors.dark3};
  border: 1px solid ${props => props.theme.colors.dark3};
  border-radius: ${props => props.theme.radii.infinite};
  transition: color ${props => props.theme.transition},
    box-shadow ${props => props.theme.transition};
  position: relative;
  cursor: pointer;
  outline: none;
  &:hover {
    background-color: ${props => props.theme.colors.dark3};
    border-color: ${props => props.theme.colors.dark3};
    box-shadow: ${props => props.theme.shadows.default};
    color: ${props => props.theme.colors.white};
  }
  &:focus {
    ${focus}
  }
  ${fontSize}
  ${size}
  ${props => props.active && activeStyle}
`

RoundButton.defaultProps = {
  size: [30, 40],
  fontSize: 'xs',
}

export default RoundButton
