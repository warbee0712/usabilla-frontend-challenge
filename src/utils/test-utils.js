import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const customRender = (node, options) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>, options)
}

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
