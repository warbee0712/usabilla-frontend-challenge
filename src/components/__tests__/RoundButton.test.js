import React from 'react'
import { render } from 'test-utils'
import RoundButton from '../RoundButton'

describe('RoundButton', () => {
  it('should match snapshot', () => {
    const { container } = render(<RoundButton>1</RoundButton>)

    expect(container.firstChild).toMatchSnapshot()
  })
  it('active should match snapshot', () => {
    const { container } = render(<RoundButton active>1</RoundButton>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
