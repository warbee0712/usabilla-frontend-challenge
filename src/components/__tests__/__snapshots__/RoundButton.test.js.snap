// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton active should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #59636b;
  border: 1px solid #59636b;
  border-radius: 9999em;
  -webkit-transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  position: relative;
  cursor: pointer;
  outline: none;
  font-size: 0.8125rem;
  height: 30px;
  width: 30px;
  background-color: #00a5c9;
  border-color: #00a5c9;
  color: #fff;
}

.c0:hover {
  background-color: #59636b;
  border-color: #59636b;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  color: #fff;
}

.c0:focus {
  border: 1px solid #00a5c9;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  outline: none;
}

.c0:hover {
  background-color: #008aab;
  border-color: #008aab;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  color: #fff;
}

<button
  class="c0"
  font-size="xs"
  size="30"
>
  1
</button>
`;

exports[`RoundButton should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #59636b;
  border: 1px solid #59636b;
  border-radius: 9999em;
  -webkit-transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  position: relative;
  cursor: pointer;
  outline: none;
  font-size: 0.8125rem;
  height: 30px;
  width: 30px;
}

.c0:hover {
  background-color: #59636b;
  border-color: #59636b;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  color: #fff;
}

.c0:focus {
  border: 1px solid #00a5c9;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  outline: none;
}

<button
  class="c0"
  font-size="xs"
  size="30"
>
  1
</button>
`;
